src/os/bluestore/NVMEDevice.cc

```diff
@@ -1140,9 +1150,7 @@ int NVMEDevice::read_random(uint64_t off, uint64_t len, char *buf, bool buffered
     queue_id = ceph_gettid();
   driver->get_queue(queue_id)->queue_task(t);
 
-  while(t->return_code > 0) {
-    t->io_wait();
-  }
+  ioc.aio_wait();
```

Comment:
I prefer t->io_wait, since one ioc may contains several tasks.

Reply:
I do not think so.  Since for read operations, we may need also split it later,  still using io_wait is not efficient.